### DISCUSSION TITLE
Improve pickle hook

### DISCUSSION THREAD

**Discussion by byehack** [pyinstaller/pyinstaller#6797] [Features & Development]
Can we add `__future__` to base_library and remove `doctest` from pickle hook?
https://github.com/pyinstaller/pyinstaller/blob/70bb887021d7c742bb6c9a6cfc510170286a4d85/PyInstaller/hooks/hook-pickle.py#L15-L16

**Comment by rokm** [comment]
We can certainly try excluding `doctest` in the `pickle` hook. 

As for `__future__`, we should check what is really going on - because this sounds like it should have been added as a hidden import somewhere else, instead of letting `doctest` (which probably directly imports `__future__`) be collected here...

**Comment by byehack** [comment]
![image](https://user-images.githubusercontent.com/26390254/166344809-b977fc7e-6896-42f1-922b-8f86c296991f.png)

script is simple `while 1: exec(input('> '))`
and compile command is 
- left: `python -m PyInstaller t.py`
- right: `python -m PyInstaller t.py --exclude-module doctest`

the `doctest` brings unnecessary modules:
![image](https://user-images.githubusercontent.com/26390254/166344938-f4258770-e99a-4a99-b67b-294f41e181f8.png)
this picture is diff-files when `doctest` excluded.(files compared after extract of PYZ archives).
you can check them: [z.zip](https://github.com/pyinstaller/pyinstaller/files/8606673/z.zip)

**Comment by rokm** [comment]
Done: #6798.
